### ISSUE TITLE
Scaling baseline and SHAPs back to original class rate

### ISSUE THREAD

**Issue by givenx** [shap/shap#1006]
I have an imbalanced dataset (positive class rate = 1%) and have downsampled the negative class to give me a 50/50 balance in the two classes.  Ignoring the challenges that comes with undersampling (loss of information), the baseline of the SHAP outputs is now close to 50% as expected.  However, I need to present the SHAP outputs including the baseline value back to the end users so that the resulting SHAP probabilities are intuitive.  Is it correct (I have a feeling it is not) to apply a straight scaling factor of 1/50 to all the SHAP values and the baseline value to bring the SHAPs in line with the original positive class rate...?  I am using lightgbm for the algorithm, not sure if that matters.

**Comment by timbayer93** [comment]
@slundberg any advice on this? Appreciate any help / guidance!

**Comment by detrin** [comment]
@givenx @timbayer93 I don't think this scales linearly. Even if you would transform shapley values based on Expected value as shown below, it is not clear **how** to transform them.
![image](https://github.com/shap/shap/assets/18104456/b9bed651-3f26-4738-a5e5-509b813146dc)

https://christophm.github.io/interpretable-ml-book/shapley.html

What I would do is that I would train the model on balanced data if you would like to do that, but evaluate the shapley values on an imbalanced dataset.

@thatlittleboy Maybe add label question. I am not sure if there is such label. I think all question-like issues should be in future redirected to Discussions
